### ISSUE TITLE
Improve rollup for Clojure applications

### ISSUE THREAD

**Issue by devn** [getsentry/sentry#5607]
Per a conversation with @mattrobenolt, @dcramer, and @bretthoerner in IRC, it was suggested that I submit an issue to improve rollup for Clojure applications by handling generated anonymous function names that appear in stack frames in the sentry stacktrace interface's hashing logic. See here: https://github.com/getsentry/sentry/blob/master/src/sentry/interfaces/stacktrace.py#L37

Here are two example raw traces which should have rolled up, but did not due to anonymous functions with generated names differing between runs of the same release of the application: https://gist.github.com/devn/521bcc000db464149a4365af1f1fdc00

**TL;DR** -- Handle anonymous Clojure function names like `myapp.foo$bar$fn__12345` and `myapp.foo$fn__98765` from breaking rollup.

**Comment by bretthoerner** [comment]
Fixed and merged.

**Comment by devn** [comment]
@bretthoerner thank you!
